### ISSUE TITLE
Not correctly reading Cgroup V2 memory limit

### ISSUE THREAD

**Issue by nickzoic** [duckdb/duckdb#15080]
### What happens?

When loading duckdb inside a docker container whose memory is limited with "--memory" option, maximum memory is set to 80% of the host's memory, not the available memory in the container.

172cfbf includes support for reading cgroups limits, but it isn't reading the limit inside the container.

### To Reproduce

```
import duckdb
print(duckdb.sql("select current_setting('memory_limit')"))
```

run this in a docker container with limited memory:

```
$ docker run -t -i --memory=16GB test:test python /script/strace-test.py
┌─────────────────────────────────┐
│ current_setting('memory_limit') │
│             varchar             │
├─────────────────────────────────┤
│ 49.9 GiB                        │
└─────────────────────────────────┘
```

It's not specific to Python either, you can get the same behaviour with:

```
duckdb -s "select current_setting('memory_limit')"
```

The host machine has 64GB RAM, so this is the right number for the host, but not for the container.

### OS:

Linux 6.8.0-49-generic x86_64

### DuckDB Version:

1.1.3

### DuckDB Client:

Python

### Full Name:

Nick Moore

### Affiliation:

Mnemote Pty Ltd

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by nickzoic** [comment]
I thought this might be specific to Docker, but upon investigation, the "\n" isn't getting trimmed from the cgroups string in ReadCGroupPath so this will affect anything using CGroups V2.

https://github.com/duckdb/duckdb/blob/ea8ad42552c8055460ae11bf3d2a882f4e9e1394/src/common/cgroups.cpp#L78-L93

Using `strace`, we can see what's going on:

```
access("/proc/self/cgroup", F_OK)       = 0
stat("/proc/self/cgroup", {st_mode=S_IFREG|0444, st_size=0, ...}) = 0
openat(AT_FDCWD, "/proc/self/cgroup", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0444, st_size=0, ...}) = 0
read(3, "0::/\n", 1023)                 = 5
close(3)                                = 0
access("/sys/fs/cgroup//\n/memory.max", F_OK) = -1 ENOENT (No such file or directory)
```

Note the `\n` left behind in the path there.  If it was gone, duckdb would be reading `/sys/fs/cgroup///memory.max` which contains the expected number for the 16GB limit.

**Comment by carlopi** [comment]
Thanks for the report & investigation!

Would you like to send a PR with the fix given you have done the detective work already?

**Comment by nickzoic** [comment]
Yeah no worries I have https://github.com/nickzoic/duckdb/tree/nickzoic/15080-cgroup-v2 , I just need to test before submitting.

**Comment by carlopi** [comment]
Fixed by #15103, thanks!.
